copy.go

```diff
@@ -112,6 +112,23 @@ type CopyGraphOptions struct {
 	// source storage to fetch large blobs.
 	// If FindSuccessors is nil, content.Successors will be used.
 	FindSuccessors func(ctx context.Context, fetcher content.Fetcher, desc ocispec.Descriptor) ([]ocispec.Descriptor, error)
+
+	// UpdateChannel is an optional channel to receive progress updates.
+	// Each update will include the number of bytes copied for a particular blob
+	// or manifest, the expected total size, and the descriptor of the blob or
+	// manifest. It is up to the consumer of the channel to differentiate
+	// between updates among different blobs and manifests; no mechanism is
+	// provided for distinguishing between them, other than the descriptor
+	// passed with each update. The total size of downloads of all blobs and
+	// manifests is not provided, as it is not known. You can calculate the
+	// percentage downloaded for a particular blob in an individual update
+	// based on the total size of that blob, which is provided in the
+	// descriptor, and the number of bytes copied, which is provided in the
+	// update.
```

Comment:
Presenting a progress bar of an operation isn't an easy job. Eventually, we will end up with an [MVC](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) pattern if we continue iterating.

Therefore, we need to define data models first for oras operations like `Copy`, `PushBytes`, etc.

Reply:
In fact, `oras` CLI has a data model so as the `buildkit`.

Reply:
/cc @qweeah

Reply:
> Presenting a progress bar of an operation isn't an easy job

True, but I am not concerned with that here. `oras` CLI might, but the `Copy()` and such library calls should not care. They only should worry about some library-centric way of publishing updates. A CLI (like `oras`) or other consumer can do with them what they want.

> we need to define data models

The data model is simple: stream of defined `struct` (or `interface`, if you prefer), each of which contains the amount of bytes transferred and descriptor to which it applies. Anything higher level would be outside the scope of this type of option, I think.